### PR TITLE
Restore call to super on ProductsController#index

### PR DESCRIPTION
**Pull request by joshcampbell**
- Fixes a bug that caused engine_storefront product grids to be empty because the original ProductsController#index method from solidus was no longer being run.
- Use Module#prepend rather than class eval to monkeypatch Spree::ProductsController so that the original index method is available
- Call `super` when responding to HTML-format requests to ProductsController#index

### CODE REVIEW

**Comment by joshcampbell** [comment]
I agree that testing is a good idea. However, I've spent a few hours today trying to get the tests running on this project and haven't figured it out yet. I may just give up for now and write the test in engine_storefront.

**Comment by bradediger** [comment]
Man, thanks for humoring me and trying to get it working. I appreciate the effort.

It's not a blocker for me at the moment. But, if it makes sense, I'd like for this project to grow to a place where testing this would be easy and obvious enough that _not_ testing it would be unacceptable. :-) I'll see if I can make any inroads on testing this.

**Comment by joshcampbell** [comment]
There's even a test over the class I modified already! But I haven't been able to get the dummy app to work. And the Gemfile pulls solidus from the master of the official github rather than the clone of solidus we refer to in engine_storefront - but when I switch to that version, I just get a different error when the dummy app tries to initialize. There's also a problem with devise needing a secret key set during initialization for some reason, and I can't seem to provide that key by simplemindedly pasting it into the environment configuration code of the dummy app...

Basically in order to proceed, I need to actually understand how spree / solidus dummy apps work.

**Comment by bradediger** [comment]
I pushed up 824bc8938c2e9d48c1ff43aa322d74b012195e4f which has a failing spec that I think starts to expose some hooks you can use to test this. For example, right now the first assertion (successful response) passes, and then the second (`@magic == 'more magic'`) fails, presumably because that's what is done in `super`. (I didn't check.)

It sounds like it may be a difficulty in your test setup? I didn't have any issue with a generated dummy app, once I figured out the key secret that _the documentation is wrong_ :man_facepalming:. My commit above also fixes the task name in the README. :-) I have two failing tests on gemini_master but have not investigated whether they're real or just artifactual.

What I think this does point out is that the control flow is getting pretty convoluted. Is it true that `super` only needs to be called for the HTML format? If so, the reason should probably be documented somewhere. If the reason is that HTML and RSS are really two different things and they'll be handled differently, well I think ya know where that one goes. :wink:

**Comment by fastjames** [comment]
It's interesting to note that the `super` call was removed in 953b1b5688 which was back in December. Has products#index been broken for that long?

**Comment by jacobeubanks** [comment]
@fastjames I don't think so. It was noticeably broken for me yesterday.

**Comment by fastjames** [comment]
The test I psuhed just now appears to cover the fix on my laptop. :crossed_fingers:

**Comment by fastjames** [comment]
That last commit fixes the extraneous failing specs. :100:

**Comment by fastjames** [comment]
Minor squash and commit reorder. No code changes.

**Comment by joshcampbell** [comment]
> It's interesting to note that the `super` call was removed in [953b1b5](https://github.com/geminimvp/solidus_product_feed/commit/953b1b56884df45cac85522dbc47aee4414a1d32) which was back in December. Has products#index been broken for that long?

Nope, it's just that Engine_storefront's Gemfile.lock had been pointing to an earlier commit in the gemini_master branch until someone finally ran bundle update and committed the result on Monday.
